.github/container/Dockerfile.pax.arm64

```diff
@@ -1,65 +1,152 @@
 # syntax=docker/dockerfile:1-labs
-###############################################################################
-## Pax for AArch64
-###############################################################################
 
 ARG BASE_IMAGE=ghcr.io/nvidia/jax:latest
-FROM ${BASE_IMAGE}
-
-# We need to build some packages from source, bring some dependencies.
-RUN apt-get update && \
-  apt-get update && \
-  apt-get install -y \
-    bat \
-    curl \
-    git \
-    gnupg \
-    rsync \
-    liblzma-dev \
-    && \
-  apt-get autoremove -y && apt-get clean && rm -rf /var/lib/apt/lists
+ARG REPO_PAXML=https://github.com/google/paxml.git
+ARG REPO_PRAXIS=https://github.com/google/praxis.git
+ARG REF_PAXML=main
+ARG REF_PRAXIS=main
+ARG SRC_PATH_PAXML=/opt/paxml
+ARG SRC_PATH_PRAXIS=/opt/praxis
 
+###############################################################################
+## build tensorflow-text and lingvo, which do not have working arm64 pip wheels
+###############################################################################
 
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE} as wheel-builder
+
+# tensorflow-text and lingvo build needs bazel
 RUN wget https://github.com/bazelbuild/bazelisk/releases/download/v1.17.0/bazelisk-linux-arm64 -O /usr/bin/bazel && \
     chmod a+x /usr/bin/bazel
 
+#------------------------------------------------------------------------------
+# build tensorflow-text 2.13.0 from source
+#------------------------------------------------------------------------------
+
+FROM wheel-builder as tftext-builder
+
+RUN <<"EOT" bash -exu
+set -o pipefail
+pip install tensorflow_datasets==4.9.2 auditwheel tensorflow==2.13.0
+git clone http://github.com/tensorflow/text.git /opt/tensorflow-text
+cd /opt/tensorflow-text
+git checkout v2.13.0
+./oss_scripts/run_build.sh
+EOT
+
+#------------------------------------------------------------------------------
+# build lingvo
+#------------------------------------------------------------------------------
+
+FROM wheel-builder as lingvo-builder
+ARG REPO_LINGVO=https://github.com/tensorflow/lingvo.git
+ARG REF_LINGVO=master
+ARG SRC_PATH_LINGVO=/opt/lingvo
+
+COPY --from=tftext-builder /opt/tensorflow-text/tensorflow_text*.whl /opt/
+
+RUN get-source.sh -f ${REPO_LINGVO} -r ${REF_LINGVO} -d ${SRC_PATH_LINGVO}
+
+# build lingvo
+RUN <<"EOT" bash -exu
+set -o pipefail
+
+pushd ${SRC_PATH_LINGVO}
+git fetch origin pull/329/head:pr329
+git cherry-pick --allow-empty pr329
+
+# Disable 2 flaky tests here
+patch -p1 <<"EOF"
```

Comment:
What do you think about using using a copy heredoc to add the patch to the container and then committing this so it doesn't look like we have unstaged files?

Reply:
This is for lingvo, which is a transitive dependency built in the `lingvo-builder` stage. The source code is not included in the later mealkit/final stages. Only the wheel is copied over.